litho1pt0/__init__.py

```diff
@@ -218,31 +228,44 @@ def property_at_lat_lon_depth_points(lat, lon, depth, quantity_ID="DENSITY"):
     lat1   = np.array(lat).reshape(-1)
     depth1 = np.array(depth).reshape(-1)
     point_properties = np.empty_like(depth1)
+    nlocations = lat1.shape[0]
 
-    layer_depths     = np.empty((nlayers, lat1.shape[0]))
-    layer_properties = np.ones((nlayers+1, lat1.shape[0])) * -99999.0  # if the point is not found it will end up in the overshoot !
-
-    # should assert here that the three arrays are equal size
+    layer_depths     = np.empty((nlayers, nlocations))
+    layer_properties = np.ones((nlayers+1, nlocations)) * -99999.0  # if the point is not found it will end up in the overshoot !
+    #should assert here that the three arrays are equal size
 
     for i in range(0, nlayers, 1 ):
         layer_depths[i], err = _interpolator.interpolate( lon1 * np.pi / 180.0, lat1 * np.pi / 180.0,
                                       _litho_data[i,l1_data_decode["DEPTH"]], order=1)
         layer_properties[i], err = _interpolator.interpolate( lon1 * np.pi / 180.0, lat1 * np.pi / 180.0,
                                       _litho_data[i,l1_data_decode[quantity_ID]], order=1)
 
-
     A = -layer_depths
     B = -depth1 * 1000.0
-    C = divmod(np.searchsorted(A.ravel(), B), A.shape[1])[0] # YEP - this seems to be the best way !!
-
-    # point_properties = np.diag(layer_properties[C[:],:])
+    th_depth = 100 # [m]
 
     point_properties = np.empty_like(depth1)
-    for i,layer in enumerate(C):
-
-        point_properties[i] = layer_properties[layer,i]
-
-    return C, point_properties.reshape(shape)
+    C = np.empty_like(depth1)
+    for i in range(0,nlocations):
+        idxs = np.where(np.abs(A[:,i]-B[i])<th_depth)[0]
+        nidxs = len(idxs)    
+        if nidxs==0:
+            idx = np.searchsorted(A[:,i], B[i]) 
+        elif nidxs ==1: 
+            idx = nidxs 
+        elif nidxs ==2:
+            if  interface_type == "TOP":
+                idx = idxs[0]
+            elif interface_type == "BOTTOM":
+                idx = idxs[1] 
+        elif nidxs >2:
+            if  interface_type == "TOP":
+                idx = idxs[0] 
+            elif interface_type == "BOTTOM":
+                idx = idxs[-1]
+        point_properties[i] = layer_properties[idx,i]   
+        C[i] = idx
+    return C.reshape(shape), point_properties.reshape(shape)
```

Comment:
Is it possible to vectorise this loop?

Reply:
Hi Ben, thanks for the suggestion, it can be vectorised, see the comment "vectorise fn property_at_lat_lon_depth_points", it does improve the speed.
![image](https://user-images.githubusercontent.com/57476594/119069791-1e9e0f80-ba2a-11eb-996c-88ed413047fb.png)

[Example for the fixed Fn property_at_lat_lon_depth_points() in Litho1.0 .txt](https://github.com/underworldcode/litho1pt0/files/6519524/Example.for.the.fixed.Fn.property_at_lat_lon_depth_points.in.Litho1.0.txt)

Reply:
If you increase the number of points by an order of magnitude, does the time difference widen?

Reply:
Yes, in the case calculated below, A.shape = [19, (1790*359)] (in the older case A.shape = [19, (179*359)] )
![image](https://user-images.githubusercontent.com/57476594/119080601-b6592900-ba3d-11eb-94ff-e4e83409e99d.png)